File: src/main/java/org/sonar/plugins/xquery/parser/visitor/XQueryAstVisitor.java
/*
 * © 2013 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.parser.visitor;

import org.sonar.plugins.xquery.language.SourceCode;
import org.sonar.plugins.xquery.parser.XQueryTree;
import org.sonar.plugins.xquery.parser.node.DependencyMapper;
import org.sonar.plugins.xquery.parser.reporter.ProblemReporter;

public interface XQueryAstVisitor
{
    public abstract void enterSource(SourceCode code, XQueryTree node, DependencyMapper mapper);

    public abstract void exitSource(XQueryTree node);

    public abstract void enterExpression(XQueryTree node);

    public abstract void exitExpression(XQueryTree node);

    public abstract void checkReport(ProblemReporter reporter);
}

File: src/main/java/org/sonar/plugins/xquery/parser/visitor/XQueryAstVisitor.kt
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.parser.visitor

import org.antlr.v4.runtime.ParserRuleContext
import org.sonar.plugins.xquery.language.SourceCode
import org.sonar.plugins.xquery.parser.node.DependencyMapper
import org.sonar.plugins.xquery.parser.reporter.ProblemReporter

interface XQueryAstVisitor {
    fun enterSource(code: SourceCode, node: ParserRuleContext, mapper: DependencyMapper)

    fun exitSource(node: ParserRuleContext)

    fun enterExpression(node: ParserRuleContext)

    fun exitExpression(node: ParserRuleContext)

    fun checkReport(reporter: ProblemReporter)
}